src/components/AddMovie.tsx

```diff
@@ -1,30 +1,162 @@
 import React from 'react';
 import styled from 'styled-components';
+import { Formik, Form, Field } from 'formik';
+import * as Yup from 'yup';
+import { ModalWindowType } from '../model';
+import { selectSelectedMovie } from '../store/moviesSlice';
+import { useAppSelector } from '../store/hooks';
+import { GENRES } from './NavBar';
 
 interface AddUpdateMovieProps {
-  onAddMovie?: any;
-  onUpdateMovie?: any;
+  type: ModalWindowType;
+  onSubmit: any;
 }
 
-const AddUpdateMovieWrapper = styled.div``;
+const AddUpdateMovieWrapper = styled.div`
+  padding: 10px;
+`;
+
+const ItemWrapper = styled.div`
+  display: flex;
+  justify-content: space-between;
+`;
+
+const ErrorMessageWrapper = styled.span`
+  color: red;
+`;
 
 const AddUpdateMovie = (props: AddUpdateMovieProps) => {
-  const onUpdateClick = () => {
-    props.onUpdateMovie(true);
-  };
+  const selectedMovie = useAppSelector(selectSelectedMovie);
 
-  const onAddMovie = () => {
-    props.onAddMovie(true);
-  };
+  const SignupSchema = Yup.object().shape({
+    title: Yup.string().required('Required'),
+    release_date: Yup.string()
+      .required('Required')
+      .matches(/\d{4}-\d{2}-\d{2}/, 'The correct format is 2022-12-12'),
```

Comment:
It's okay to store date as string internally, so no changes are needed.
But in production projects it's usually supposed to store dates as `Date` object because it's easier to work with date when it's in Date type. For example, validation is much easier in this case: `date instanceof Date`, and different date mutations are much easier(including formatting and changing format of date).